text/0000-add-geo-processing-abilities-to-skylark.md

```diff
@@ -0,0 +1,535 @@
+PostGIS. The API is following that of the Shapley python library so should be familiar to most geospatial 
+python developers.
+<!-- - Why is this design the best in the space of possible designs?
+- What other designs have been considered and what is the rationale for not choosing them?
+- What is the impact of not doing this? -->
+
+
+
+# Prior art
+[prior-art]: #prior-art
+
+- [PostGIS Functions](https://postgis.net/docs/reference.html#Geometry_Editors) : A breakdown of geospatial functions that PostGIS uses (this is probably the most comprehensive list of these kinds of functions)
+- [golang-geo](https://github.com/kellydunn/golang-geo) : A geospatial library for Go
+- [https://github.com/golang/geo ](https://github.com/golang/geo) : Googles S2 library implement in Go
+- [Shapley](https://pypi.org/project/Shapely/) : One of the core python libraries for geometries, main inspiration for the API here.
+- [GeoPandas](http://geopandas.org/) : DataFrame implementation that has geospatial functionality.
+
+<!-- Discuss prior art, both the good and the bad, in relation to this proposal.
+A few examples of what this can include are:
+
+- Does this feature exist in other places and what experience have their community had?
+- For community proposals: Is this done by some other community and what were their experiences with it?
+- For other teams: What lessons can we learn from what other communities have done here?
+- Papers: Are there any published papers or great posts that discuss this? If you have some relevant papers to refer to, this can serve as a more detailed theoretical background.
+
+This section is intended to encourage you as an author to think about the lessons from other projects, provide readers of your RFC with a fuller picture.
+If there is no prior art, that is fine - your ideas are interesting to us whether they are brand new or if it is an adaptation from other languages.
+
+Note that while precedent set by other projects is some motivation, it does not on its own motivate an RFC.
+Please also take into consideration that Qri sometimes intentionally diverges from other projects. -->
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+There are a bunch of features that this proposal doesn't include, is it valid to not include these in the first pass:
+- Projections : Geospatial data can be projected in to many different local coordinate systems which are 
+useful in different contexts. We aren't including this in this first implementation but if it's important we could include it
+- Geometry modification methods : Translate, Rotate etc functions 
+
+Other questions:
+
+- Do we want this to help support the visualization part of Qri when data is geospatial?
+- What volumes of data should this be looking to support? Do we need to be building and somehow persisting with the dataset geospatial indexes that allow for faster processing / querying? 
```

Comment:
For now I like the notion of focusing on what can fit into an in-memory byte array, so roughly 1-2Gig limit. 

For in-memory representations we're taking a good hard look at [Apache Arrow](https://arrow.apache.org/) to see if that can't bake in native support, forming the underlying starlark data representation while also giving us a serialization-free switch to pandas & R

On the streaming side, We need to ship an RFC that connects our lower level `dsio.EntryReader/dsio.EntryWriter` primitives up into a reader/writer "File-Like Object" in starlark to handle larger data volumes. We're currently working out how to square this stream-oriented processing with CBOR's variable-length data types and this Arrow stuff. That bit is going to take a while, but is the long term goal.

 _Lots_ more to come on the subject of data volume in the next few months. We'll be structuring all of these conversations as RFCs.